frontend/src/utils/misc.js

```diff
@@ -0,0 +1,37 @@
+const snakeToCamel = str => str.replace(/_(.)/g, s => s[1].toUpperCase());
+
+export const renameObjectKeys = obj => {
+    if (typeof obj !== "object" || obj === null) {
+        return obj;
+    }
+
+    if (Array.isArray(obj)) {
+        return obj.map(o => renameObjectKeys(o));
+    }
+
+    return Object.fromEntries(Object.entries(obj).map(([key, val]) => [snakeToCamel(key), renameObjectKeys(val)]));
+};
+
+/**
+ * Extract the year, month date from datetime string
+ * Not using javascript's Date class cause it sucks at datetime string parsing
+ * @param {string} s ISO Datetime string in format of `YYYY-MM-DDTHH:MM:SS`
+ */
+export const getDateFromISOString = s => s.split("T")[0].split("-");
+export const getTimeFromISOString = s =>
+    s
+        .split("T")[1]
+        .split(":")
+        .map(i => i.split(".")[0]);
+
+export const getLocalDateTimeFromISOString = s => {
```

Comment:
_I'm not sure about using this function in redux level_.

I thought the intention was to store the `Date` instance in redux instead of an array converted string. I think we should do one of two tings:

1. Store the `Date` object of `writeTime` in redux. And write the util in a way similar to Python `datetime.strftime`.
2. Store the array of UTC time strings as previously is. And use this function in the component level.